### PR TITLE
Cross build with sbt 2.0.0-M2

### PR DESCRIPTION
**Pull request by agboom**
Hi, this is still a draft and my first attempt at migration. I was looking to contribute to the [sbt 2.x plugin migration](https://github.com/sbt/sbt/wiki/sbt-2.x-plugin-migration) and found sbt-git to be one that's used often.

I had to disable sbt-ci-release for now, because it doesn't support 2.x yet and [uses sbt-git itself as well](https://github.com/sbt/sbt-ci-release/blob/main/project/plugins.sbt). So I think we need to do some kind of bootstrap without sbt-ci-release before we can migrate that one too?

Let me know what you think. 

Edit: apparently sbt-ci-release was migrated to 2.x just now, so I'll update my PR accordingly

Edit: it seems to be unable to resolve sbt-ci-release `1.6.1+13-1458cb0b-SNAPSHOT`, I'm probably missing something here

### CODE REVIEW

File: src/main/scala/com/github/sbt/git/ConsoleGitRunner.scala [hunk 1]
```diff
@@ -17,15 +19,15 @@ object ConsoleGitRunner extends GitRunner {
 
   // in order to enable colors we trick git into thinking we're a pager, because it already knows we're not a tty
   val colorSupport: Seq[(String, String)] =
-    if(ConsoleLogger.formatEnabled) Seq("GIT_PAGER_IN_USE" -> "1")
+    if(ConsoleAppender.formatEnabledInEnv) Seq("GIT_PAGER_IN_USE" -> "1")
```
Comment (agboom) [inline src/main/scala/com/github/sbt/git/ConsoleGitRunner.scala:22]: Both methods are deprecated, the latter points to `Terminal.isColorEnabled`, but that one's private. I'm not sure if these will be removed in the final 2.x, but for now I couldn't find a suitable replacement.

File: src/sbt-test/git-versioning/find-tag-newest/test [hunk 2]
```diff
@@ -1,6 +1,8 @@
 > git init
 > git config user.email "test@jsuereth.com"
 > git config user.name "Tester"
+> git config core.hookspath /dev/null
+> git config commit.gpgsign false
```
Comment (agboom) [inline src/sbt-test/git-versioning/find-tag-newest/test:5]: Some scripted tests started to take a very long time. I figured it was wainting for a GPG password prompt, so I added these lines I added before in a plugin of my own.

**Comment by mzuehlke** [comment]
there is #256

**Comment by agboom** [comment]
Let's continue in #256

**Pull request closed without merging**